### PR TITLE
feat(access-tokens): expose personal/project/group token management

### LINKED ISSUE #441: feat(access-tokens): expose project/group/personal access tokens management
👍 React to this issue if you need this feature — helps us prioritise.

## Problem

Access tokens (PATs, project access tokens, group access tokens) are core CI/automation primitives. We expose token **scope detection** for the running token (`TokenScopeDetector`) but no tools to list/rotate/revoke them at user/project/group level.

## Investigation done (current state)

Verified against `src/entities/`:

- ✅ `services/TokenScopeDetector.ts` introspects the CURRENT token via `/api/v4/personal_access_tokens/self` — good for whoami
- ❌ `grep -ri 'access_token' src/entities/` returns **0 hits**
- ❌ No tool lists tokens belonging to a project/group/user

**Re-verify before coding:**

```bash
grep -rIni 'access.*token' src/entities/
ls src/entities/ | grep token
```

## Acceptance criteria

- New entity `src/entities/access_tokens/` with:
  - `browse_access_tokens` actions: `list_personal` (self), `list_project`, `list_group`, `get`
  - `manage_access_token` actions: `create_project`, `create_group`, `rotate` (returns new token value once), `revoke`
- Token-value responses must be flagged sensitive in tool result metadata
- Integration test creating a PAT for a test project, rotating it, revoking it
- Permission gating: requires `api` scope; owner+admin for project/group tokens

## GitLab API

- Endpoints:
  - `/personal_access_tokens`, `/personal_access_tokens/self/rotate`
  - `/projects/:id/access_tokens`, `/projects/:id/access_tokens/:token_id/rotate`
  - `/groups/:id/access_tokens`, `/groups/:id/access_tokens/:token_id/rotate`
- Tier: Free
- Docs: https://docs.gitlab.com/api/project_access_tokens/, .../group_access_tokens/, .../personal_access_tokens/

## Estimate

6h

## Context

Found during GitLab 18→19 API landscape analysis. Token rotation is a common compliance need.

## Consolidation decision

- **Tools:** one CQRS pair — `browse_access_tokens` + `manage_access_token`. The three scopes (personal/project/group) fold in as actions, not three tools.
  - `browse_access_tokens`: `list_personal`, `list_project`, `list_group`, `get`
  - `manage_access_token`: `create_project`, `create_group`, `rotate`, `revoke`
- **Feature flag:** gated by `USE_ACCESS_TOKENS` (added in #456).
- **Profiles (enabled):** devops (rotates CI tokens), security, admin, full-access, readonly (read-only). Disabled elsewhere — token CRUD is sensitive/admin-level.
- **Tier:** Free. Token-value responses flagged sensitive in result metadata. Project/group create requires owner+admin.

Depends on #456 (flag + profile plumbing).

### PR DESCRIPTION
**Pull request by polaz**
## Summary

Adds the `browse_access_tokens` / `manage_access_token` CQRS pair so agents can inspect, create, rotate, and revoke GitLab access tokens (the core CI/automation credentials).

- **browse_access_tokens** (Query): `list_personal` (current user's PATs; admins may filter by `user_id`), `list_project`, `list_group`, `get`
- **manage_access_token** (Command): `create_project`, `create_group`, `rotate`, `revoke`

## Implementation notes

- **REST-only by necessity.** GitLab exposes no GraphQL surface for personal/project/group access tokens (verified by live schema introspection: zero mutation or query fields matching `/token/i`). All actions map to the `/personal_access_tokens`, `/projects/:id/access_tokens` and `/groups/:id/access_tokens` endpoints.
- The three scopes fold in as actions, not separate tools. `get` / `rotate` / `revoke` infer the scope from `project_id` / `group_id` (neither => personal); a schema refine forbids passing both.
- `create_*` and `rotate` return a token value exactly once; those responses are wrapped with a `_meta.sensitive` marker and a store-securely notice so the secret is flagged in the tool output.
- Personal-token creation is intentionally absent: GitLab only allows it for admins against a specific user, out of scope for this tool.
- Token scopes are accepted as a free-form string array (not a fixed enum) so new GitLab scopes work without a code change.
- Gated behind `USE_ACCESS_TOKENS`; token-scope requirements (`browse` => `api`/`read_api`, `manage` => `api`) wired into `TokenScopeDetector`.

## Testing

- Unit: 20 tests covering every browse + manage action, scope routing (personal/project/group), the sensitive-metadata flag, `token_id` coercion, and the mutual-exclusion / empty-scopes schema guards (REST `enhancedFetch` mocked).
- Integration: schema validation + `list_personal` end-to-end + env-adaptive project-token lifecycle (create -> rotate -> revoke) that runs only with a writable `test/` project and always revokes the created token in cleanup.
- Full suite green: 5076 unit tests, 424 integration tests, clean build and lint.

Closes #441

## Summary by CodeRabbit

* **New Features**
  * Added access token management: list personal/project/group tokens; create, rotate, and revoke tokens. Creation/rotation return one-time sensitive tokens with a one-time retrieval notice. Availability is controlled by configuration.

* **Validation**
  * Added input validation for all access-token actions with safeguards against invalid parameter combinations.

* **Tests**
  * Added comprehensive unit and integration tests covering schemas, routing, lifecycle (create → rotate → revoke), and edge cases.

### CODE REVIEW

File: src/entities/access_tokens/schema.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+const accessLevelField = z
+  .union([z.literal(10), z.literal(20), z.literal(30), z.literal(40), z.literal(50)])
+  .optional()
+  .describe('Access level: 10 Guest, 20 Reporter, 30 Developer, 40 Maintainer, 50 Owner.');
+
+const expiresAtField = z
+  .string()
+  .optional()
+  .describe('Expiry date in YYYY-MM-DD format (e.g. "2026-12-31").');
+
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/entities/access_tokens/schema.ts:38]: @coderabbitai Fixed - scopes elements are now `z.string().trim().min(1)` (rejects empty/whitespace entries) and expires_at is validated against `/^\d{4}-\d{2}-\d{2}$/`. Added unit tests asserting both rejections.

File: src/services/TokenScopeDetector.ts [hunk 2]
```diff
@@ -143,6 +143,10 @@ const TOOL_SCOPE_REQUIREMENTS: Record<string, GitLabScope[]> = {
   browse_registry: ['api', 'read_api', 'read_registry'],
   manage_registry: ['api', 'write_registry'],
 
+  // Access tokens (personal/project/group) - REST-only, require full api scope
+  browse_access_tokens: ['api', 'read_api'],
+  manage_access_token: ['api'],
```
[the remark this answers is not in the document]
Comment (polaz) [reply src/services/TokenScopeDetector.ts]: @coderabbitai Fixed - comment now reads 'browse requires api/read_api; manage requires api', matching the requirements map.

File: tests/integration/schemas/access-tokens.test.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+        const rotated = (await helper.executeTool('manage_access_token', {
+          action: 'rotate',
+          project_id: testProjectPath,
+          token_id: tokenId,
+        })) as TokenResponse & { _meta?: { sensitive?: boolean } };
+        // Rotation issues a NEW token id; track that one for cleanup.
+        tokenId = rotated.id;
+        expect(typeof rotated.token).toBe('string');
+        console.log(`  rotated project token on ${testProjectPath}`);
```
[the remark this answers is not in the document]
Comment (polaz) [reply tests/integration/schemas/access-tokens.test.ts:128]: @coderabbitai Fixed - added `expect(rotated._meta?.sensitive).toBe(true)` after the rotation block, mirroring the create assertion.
